Swerve/SwerveTemplates/VisionBaseSwerve.java

```diff
@@ -75,16 +77,25 @@ public void updateVisionPoses() {
      * The input should be relative to the robot, aka a Transform2d input with x = +1, y = -1, rot = pi 
      * should move the robot 1 unit forward, 1 unit left, perform a 180 degree rotation
      * @param driverInput a Transform2d of the driver-desired movement of the robot, relative to the robot
+     * also remember that rotation is in radians and x and y should be in meters
      * @return an adjusted vector that will nudge the robot closer to the detected object
      */
     public Transform2d addVisionMovementAdjustment(Transform2d driverInput) {
-        Optional<DetectedObject> detection = limelight.getLastestDetection();
-        if (detection.isPresent()) {
+        DetectedObject bestDetection = null;
+        double bestArea = -1;
+        for (Limelight limelight : limelights) {
+            Optional<DetectedObject> detection = limelight.getLastestDetection();
+            if (detection.isPresent() && detection.get().area > bestArea) {
```

Comment:
could a null pointer except occur on line 96, 97, and 98 due to this conditional?
idk if the detection area can be less than -1 in any circumstance (like no detection, etc). If there's no chance of that, then it should be good

Reply:
Gotcha, sounds good!